diff --git Sluift/Console.h Sluift/Console.h
index d1687bf..6ed6f58 100644
--- Sluift/Console.h
+++ Sluift/Console.h
@@ -8,7 +8,7 @@
 
 #include <string>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 
 #include <Swiften/Base/Override.h>
 
diff --git Sluift/ITunesInterface.h Sluift/ITunesInterface.h
index a2cd06a..698e6ed 100644
--- Sluift/ITunesInterface.h
+++ Sluift/ITunesInterface.h
@@ -9,7 +9,7 @@
 #include <memory>
 #include <string>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 
 #include <Swiften/Base/API.h>
 
diff --git Sluift/Terminal.h Sluift/Terminal.h
index 9d44095..3443d16 100644
--- Sluift/Terminal.h
+++ Sluift/Terminal.h
@@ -8,7 +8,7 @@
 
 #include <string>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 
 namespace Swift {
     class Completer;
diff --git Swiften/Elements/Stanza.h Swiften/Elements/Stanza.h
index 9a69696..08fa4b1 100644
--- Swiften/Elements/Stanza.h
+++ Swiften/Elements/Stanza.h
@@ -11,7 +11,7 @@
 #include <vector>
 
 #include <boost/date_time/posix_time/ptime.hpp>
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 
 #include <Swiften/Base/API.h>
 #include <Swiften/Elements/ToplevelElement.h>
diff --git Swiften/FileTransfer/FileTransferTransporter.h Swiften/FileTransfer/FileTransferTransporter.h
index 5af830c..e0ffc87 100644
--- Swiften/FileTransfer/FileTransferTransporter.h
+++ Swiften/FileTransfer/FileTransferTransporter.h
@@ -8,7 +8,7 @@
 
 #include <vector>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 #include <boost/signals2.hpp>
 
 #include <Swiften/Base/API.h>
diff --git Swiften/FileTransfer/IBBReceiveSession.h Swiften/FileTransfer/IBBReceiveSession.h
index 11dfb63..797394c 100644
--- Swiften/FileTransfer/IBBReceiveSession.h
+++ Swiften/FileTransfer/IBBReceiveSession.h
@@ -8,7 +8,7 @@
 
 #include <memory>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 #include <boost/signals2.hpp>
 
 #include <Swiften/Base/API.h>
diff --git Swiften/JID/JID.h Swiften/JID/JID.h
index 302e37e..dc92f53 100644
--- Swiften/JID/JID.h
+++ Swiften/JID/JID.h
@@ -9,7 +9,7 @@
 #include <iosfwd>
 #include <string>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 
 #include <Swiften/Base/API.h>
 
diff --git Swiften/Network/BoostConnectionServer.h Swiften/Network/BoostConnectionServer.h
index 41fdbe2..3dd9830 100644
--- Swiften/Network/BoostConnectionServer.h
+++ Swiften/Network/BoostConnectionServer.h
@@ -10,7 +10,7 @@
 
 #include <boost/asio/io_service.hpp>
 #include <boost/asio/ip/tcp.hpp>
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 #include <boost/signals2.hpp>
 
 #include <Swiften/Base/API.h>
diff --git Swiften/Network/ConnectionServer.h Swiften/Network/ConnectionServer.h
index 0c8a000..769ab9f 100644
--- Swiften/Network/ConnectionServer.h
+++ Swiften/Network/ConnectionServer.h
@@ -8,7 +8,7 @@
 
 #include <memory>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 #include <boost/signals2.hpp>
 
 #include <Swiften/Base/API.h>
diff --git Swiften/Parser/AttributeMap.h Swiften/Parser/AttributeMap.h
index 798649a..804d6aa 100644
--- Swiften/Parser/AttributeMap.h
+++ Swiften/Parser/AttributeMap.h
@@ -10,7 +10,7 @@
 #include <string>
 #include <vector>
 
-#include <boost/optional/optional_fwd.hpp>
+#include <boost/optional/optional.hpp>
 
 #include <Swiften/Base/API.h>
 #include <Swiften/Parser/Attribute.h>
