diff --git krusader/krslots.cpp krusader/krslots.cpp
index 336b349..8bee119 100644
--- krusader/krslots.cpp
+++ krusader/krslots.cpp
@@ -230,8 +230,8 @@ void KRslots::compareContent(QUrl url1, QUrl url2)
 
     // kdiff3 sucks with spaces
     if (QUrl::fromLocalFile(diffProg).fileName() == "kdiff3" && !url1.toDisplayString().contains(" ") && !url2.toDisplayString().contains(" ")) {
-        tmp1 = url1.toDisplayString();
-        tmp2 = url2.toDisplayString();
+        tmp1 = url1.path();
+        tmp2 = url2.path();
     } else {
         if (!url1.isLocalFile()) {
             if (!downloadToTemp(url1, tmp1)) {
